File: baek_11497.cpp
#include <stdio.h>
#include <algorithm>
#include <math.h>

using namespace std;

int main(void) {
	int t, n, idx, idx1, answer, tmp;
	int arr[10000];
	int arr1[10000];
	scanf("%d", &t);
	for(int tcase=0; tcase<t; tcase++){
		answer = -1;
		scanf("%d", &n);
		for(int i=0; i<n; i++)
			scanf("%d", &arr[i]);
		sort(arr, arr+n);
		tmp = 0;
		
		for(idx=0; idx<n; idx+=2){
			arr1[tmp] = arr[idx];
			tmp++;
		}
		if(idx==n)
			idx = n-1;
		else
			idx = n-2;
		for( ; idx>0; idx-=2){
			arr1[tmp] = arr[idx];
			tmp++;
		}
		for(int i=0; i<n-1; i++)
			if(abs(arr1[i]-arr1[i+1])>answer)
				answer = abs(arr1[i]-arr1[i+1]);
		printf("%d\n", answer);
	}
    return 0;
}

File: baek_11497.java
package com.company;

import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int[] arr0 = new int[100000];
        int[] arr1 = new int[100000];
        int n, t, ans, idx0, idx1;
        t = sc.nextInt();
        for(int tcase=0; tcase<t; tcase++){
            n = sc.nextInt();
            ans = -1;
            idx0 = 0;
            idx1 = 0;
            for(int i=0; i<n; i++)
                arr0[i] = sc.nextInt();
            Arrays.sort(arr0, 0, n);
            for( ; idx0<n; idx0+=2){
                arr1[idx1] = arr0[idx0];
                idx1++;
            }
            if(idx0==n)
                idx0 = n-1;
            else
                idx0 = n-2;
            for( ; idx0>0; idx0-=2){
                arr1[idx1] = arr0[idx0];
                idx1++;
            }
            for(int i=0; i<n-1; i++) {
                int abs = Math.abs(arr1[i] - arr1[i + 1]);
                if(abs > ans)
                    ans = abs;
            }
            System.out.println(ans);
        }
    }
}
